packages/dd-trace/test/appsec/standalone.spec.js

```diff
@@ -106,4 +154,245 @@ describe('Appsec Standalone', () => {
+
+      span._spanContext._sampling = {
+        priority: USER_KEEP,
+        mechanism: SAMPLING_MECHANISM_APPSEC
+      }
+
+      span._spanContext._trace.tags[APPSEC_PROPAGATION_KEY] = '1'
+
+      const carrier = {}
+      const propagator = new TextMapPropagator(config)
+      propagator.inject(span._spanContext, carrier)
+
+      assert.property(carrier, 'x-datadog-trace-id')
+      assert.property(carrier, 'x-datadog-parent-id')
+      assert.property(carrier, 'x-datadog-sampling-priority')
+      assert.propertyVal(carrier, 'x-datadog-tags', '_dd.p.appsec=1')
+    })
+
+    it('should not reset priority if standalone disabled', () => {
+      delete config.appsec.standalone
+      standalone.configure(config)
+
+      const span = new DatadogSpan(tracer, processor, prioritySampler, {
+        operationName: 'operation'
+      })
+
+      span._spanContext._sampling = {
+        priority: USER_KEEP,
+        mechanism: SAMPLING_MECHANISM_APPSEC
+      }
+
+      const carrier = {}
+      const propagator = new TextMapPropagator(config)
+      propagator.inject(span._spanContext, carrier)
+
+      assert.property(carrier, 'x-datadog-trace-id')
+      assert.property(carrier, 'x-datadog-parent-id')
+      assert.property(carrier, 'x-datadog-sampling-priority')
+    })
+
+    it('should clear tracestate datadog info', () => {
+      standalone.configure(config)
+
+      const span = new DatadogSpan(tracer, processor, prioritySampler, {
+        operationName: 'operation'
+      })
+
+      span._spanContext._sampling = {
+        priority: USER_KEEP,
+        mechanism: SAMPLING_MECHANISM_APPSEC
+      }
+
+      const tracestate = new TraceState()
+      tracestate.set('dd', 't.tid:666b118100000000;t.dm:-1;s:1;p:73a164d716fcddff')
+      tracestate.set('other', 'id:0xC0FFEE')
+      span._spanContext._tracestate = tracestate
+
+      const carrier = {}
+      const propagator = new TextMapPropagator(config)
+      propagator.inject(span._spanContext, carrier)
+
+      assert.propertyVal(carrier, 'tracestate', 'other=id:0xC0FFEE')
+    })
+  })
+
+  describe('StandaloneASMPriorityManager', () => {
+    let prioritySampler
+    let tags
+    let context
+
+    beforeEach(() => {
+      tags = { 'manual.keep': 'true' }
+      prioritySampler = new standalone.StandAloneAsmPrioritySampler('test')
+
+      context = {
+        _sampling: {},
+        _trace: {
+          tags: {}
+        }
+      }
+      sinon.stub(prioritySampler, '_getContext').returns(context)
+    })
+
+    describe('_getPriorityFromTags', () => {
+      it('should keep the trace if manual.keep and _dd.p.appsec are present', () => {
+        context._trace.tags[APPSEC_PROPAGATION_KEY] = 1
+        assert.strictEqual(prioritySampler._getPriorityFromTags(tags, context), USER_KEEP)
+        assert.strictEqual(context._sampling.mechanism, SAMPLING_MECHANISM_APPSEC)
+      })
+
+      it('should return undefined if manual.keep or _dd.p.appsec are not present', () => {
```

Comment:
Why is this test working? in `tags` manual.keep is defined: `{ 'manual.keep': 'true' }`

Reply:
but not `_dd.p.appsec`, both tags must be present in order to get `USER_KEEP` priority